scripts/bootloader/zb_ota_sign.py

```diff
@@ -0,0 +1,296 @@
+        zigbee_comment = ''
+    elif len(zigbee_comment) > 30:
+        click.echo('Warning: truncating the comment to 30 bytes.')
+        zigbee_comment = zigbee_comment[:30]
+
+    if zigbee_manufacturer_id is None:
+        zigbee_manufacturer_id = 0xFFFF
+
+    if zigbee_image_type is None:
+        zigbee_image_type = 0xFFFF
+
+    # Set the external_app to false in --zigbee is set
+    inner_external_app = external_app
+    if zigbee:
+        inner_external_app = False
+
+    if zigbee_ota_min_hw_version is not None and zigbee_ota_min_hw_version > 0xFFFF:
+        raise click.BadParameter('Exceeds 2-byte long integer.', param_hint='zigbee-ota-min-hw-version')
+
+    if zigbee_ota_max_hw_version is not None and zigbee_ota_max_hw_version > 0xFFFF:
+        raise click.BadParameter('Exceeds 2-byte long integer.', param_hint='zigbee-ota-max-hw-version')
+
+    if zigbee and (hw_version > 0xFFFF):
+        raise click.BadParameter('Exceeds 2-byte long integer.', param_hint='hw-version')
+
+    # Warn user if minimal/maximum zigbee ota hardware version are not correct:
+    #   * only one of them is given
+    #   * minimum version is higher than maximum version
+    #   * hw_version is inside the range specified by minimum and maximum hardware version
+    if (type(zigbee_ota_min_hw_version) is int) != (type(zigbee_ota_max_hw_version) is int):
+        click.echo('Warning: min/max zigbee ota hardware version is missing. Discarding min/max hardware version.')
+    elif type(zigbee_ota_min_hw_version) is int:
+        if zigbee_ota_min_hw_version > zigbee_ota_max_hw_version:
+            click.echo('Warning: zigbee-ota-min-hw-version is higher than zigbee-ota-max-hw-version.')
+        if (hw_version > zigbee_ota_max_hw_version) or (hw_version < zigbee_ota_min_hw_version):
+            click.echo('Warning: hw-version is outside the specified range specified by zigbee_ota_min_hw_version and zigbee_ota_max_hw_version.')
+
+    # Generate a DFU package. If --zigbee is set this is the inner DFU package
+    # which will be used as a binary input to the outer DFU package
+    package = Package(hw_version,
+                      application_version_internal,
+                      application,
+                      inner_external_app,
+                      zigbee,
+                      zigbee_manufacturer_id,
+                      zigbee_image_type,
+                      zigbee_comment,
+                      zigbee_ota_min_hw_version,
+                      zigbee_ota_max_hw_version)
+
+    package.generate_package(zipfile_path)
+
+    if zigbee:
+        from shutil import copyfile
+        from os import remove
+
+        log_message = "Zigbee update created at {0}".format(package.zigbee_ota_file.filename)
+        click.echo(log_message)
+
+        # Taking the inner Zigbee package as input for the outer DFU package
+        binfile = package.zigbee_ota_file.filename.replace(".zigbee", ".bin")
```

Comment:
In theory there can be `.zigbee` string inside filename (not in the end of a file) and these occurrences will be also replaced

Reply:
fixed